### PR TITLE
fix: parsing of string map values

### PR DESCRIPTION
**Pull request by jake-bassett**
## Description
Fix string map value parsing by only splitting on the first delimiter.

### CODE REVIEW

File: projects/components/src/filtering/filter/filter-delimiters.ts [hunk 1]
```diff
@@ -1,3 +1,11 @@
+import { isNonEmptyString } from '@hypertrace/common';
+
 export const ARRAY_DELIMITER = ',';
 export const MAP_LHS_DELIMITER = '.';
 export const MAP_RHS_DELIMITER = ':';
+
+export const splitFirstOccurrence = (str: string, delimiter: string): string[] => {
+  const firstIndex = str.indexOf(delimiter);
+
+  return [str.substr(0, firstIndex), str.substr(firstIndex + 1)].filter(isNonEmptyString);
```
Comment (aaron-steinfeld) [inline projects/components/src/filtering/filter/filter-delimiters.ts:10]: nit - I don't think this handles edge cases correctly (at least anything where a empty string is part of the expected split result such as a string that starts or ends with the delimiter, or an empty string)

File: projects/components/src/filtering/filter/parser/types/contains-filter-parser.ts [hunk 2]
```diff
@@ -46,7 +46,7 @@ export class ContainsFilterParser extends AbstractFilterParser<PossibleValuesTyp
     splitFilter: SplitFilter<FilterOperator>
   ): string[] | undefined {
     if (splitFilter.lhs === attribute.displayName) {
-      return splitFilter.rhs.split(MAP_RHS_DELIMITER);
+      return splitFirstOccurrence(splitFilter.rhs, MAP_RHS_DELIMITER);
```
Comment (aaron-steinfeld) [inline projects/components/src/filtering/filter/parser/types/contains-filter-parser.ts:49]: Could we add a test for this?

File: projects/components/src/filtering/filter/filter-delimiters.ts (hunk 1 above)
Comment (jake-bassett) [reply projects/components/src/filtering/filter/filter-delimiters.ts:10]: Added a test for that case.

Comment (aaron-steinfeld) [reply projects/components/src/filtering/filter/filter-delimiters.ts:10]: the test cases look fine, but doesn't this still behave differently than split itself? If that's acceptable for how it's used then it doesn't need to change, but should be local to the file it's used vs exported like a generic util.

Comment (aaron-steinfeld) [reply projects/components/src/filtering/filter/filter-delimiters.ts:10]: Just noticed the name changed in this rev - that's better, but I think it'd be cleaner to just keep it local

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**